### PR TITLE
BaSP-TG-80: Update Employee Forms

### PR DESCRIPTION
**Pull request by iaracriscenti**
### Task's brief description:
Implement *React Hook Form* on Employee Add and Edit Forms to improve its operation and behavior, adding the validations to each type of input and making sure that the corresponding styles are seen.

### **Please test in a 1440x900 resolution**

### CODE REVIEW

File: src/Components/Admin/Employees/Form/index.jsx [hunk 1]
```diff
@@ -1,167 +1,208 @@
...
-  };
-
-  const onChangePM = (e) => {
-    setUserInput({ ...userInput, isProjectManager: e.target.checked });
-  };
-
-  const onSubmit = () => {
+  const onSubmit = (data) => {
     if (edit) {
+      console.log(data);
```
Comment (NicolasFontana) [inline src/Components/Admin/Employees/Form/index.jsx:81]: Should delete this console.log

File: src/Components/Admin/Employees/Form/index.jsx [hunk 2]
```diff
@@ -1,167 +1,208 @@
...
+        data.phone === item.phone &&
+        data.email === item.email &&
+        data.password === item.password &&
+        data.active === item.active &&
+        data.isProjectManager === item.isProjectManager &&
+        data.projects === item.projects.map((x) => x._id).toString() &&
+        data.timeSheets === item.timeSheets.map((x) => x._id).toString()
       ) {
-        setResponse({ message: "There haven't been any changes", data: {}, error: true });
+        setResponse({ message: "There haven't been string changes", data: {}, error: true });
```
Comment (NicolasFontana) [inline src/Components/Admin/Employees/Form/index.jsx:93]: The user may not know what you mean by "string", I think "There haven't been any changes" is better.

**Review by NicolasFontana — CHANGES_REQUESTED** [review]

**Review by Johsua30 — APPROVED** [review]
Messages and validations are looking good

**Review by NicolasFontana — APPROVED** [review]
Nice!

**Comment by tinchorova** [comment]
## Test results
Tested locally on: https://alexis-trackgenix-app-git-feature-tg-80-updat-0e9869-basp-m2022.vercel.app/admin/employees

**Test results: Failed** ⚠️ 

bugs found: 
1. Validations should have an 'on blur' event listener
2. When writing an input, the required message is moving the inputs
image
3. When the user focuses on an input to correct the data it should not be in red with the invalid message because he/she is trying to correct it.
image
4. The invalid message shouldn't appear instantly when the user inputs a wrong data, (this is connected to point 1.)
image
image
5. Point .4 applies here too. Invalid messages should change when 'on blur'
image
image
6. Wrong validation on email, invalid input is taken as correct (it does not pass when clicking on edit button but there is no error message)
image
7. Why is an admin allowed to change employee's password?
image

**Review by tinchorova — CHANGES_REQUESTED** [review]
Please refer to: https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/40#issuecomment-1166053779

**Comment by iaracriscenti** [comment]
> Please refer to: [#40 (comment)](https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/40#issuecomment-1166053779)

### BUGS:
- 1, 4, 5 *FIXED* --> changed useForm mode, now validations display  when on blur event occurs
- 2 *FIXED* --> changed modal form styles to allow more space
- 3 *FIXED* --> changed component on focus pseudo-class
- 6 *FIXED* --> changed email validation
- 7 *Will be taken into account in the coming weeks*

**Comment by tinchorova** [comment]
## Test results
Tested locally on: https://alexis-trackgenix-app-git-feature-tg-80-updat-0e9869-basp-m2022.vercel.app/admin/employees

**Test results:** Approved, all functions are working, **but take this in count:**

bugs found:
1. When writing in password field the input moves
image
image
image

**Review by tinchorova — APPROVED** [review]
Good job! **important** please refer to https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/40#issuecomment-1166379603

**Pull request merged**